File: app/src/main/java/br/com/faru/a99challenge/presentation/person/PersonsContract.java
package br.com.faru.a99challenge.presentation.person;

import java.util.List;

import br.com.faru.a99challenge.model.Person;

public interface PersonsContract {

    interface View {

        void showResults(List<Person> persons);

        void setProgressIndicator(boolean active);

        void goToPerson(Person person);

        void showMessage(String message);
    }

    interface UserInteractions {

        void setView(View view);

        void onCreate();

        void onClick(Person person);
    }

}

File: app/src/main/java/br/com/faru/a99challenge/presentation/person/PersonsContract.kt
package br.com.faru.a99challenge.presentation.person

import br.com.faru.a99challenge.model.Person

interface PersonsContract {

    interface View {

        fun showResults(persons: List<Person>)

        fun setProgressIndicator(active: Boolean)

        fun goToPerson(person: Person)

        fun showMessage(message: String)
    }

    interface UserInteractions {

        fun setView(view: View)

        fun onActivityCreated()

        fun onClick(person: Person)
    }

}
